### ISSUE TITLE
Freezes VS and then crashes when making any changes to a bundleconfig.json file

### ISSUE THREAD

**Issue by CoolDadTx** [madskristensen/BundlerMinifier#355]
### Installed product versions
- Visual Studio: 2017 Enterprise, 15.7.0 Preview 4.0
- 2.6.378

### Description
I have a bundleconfig.json file in a solution that I'm playing around with. If I make any change to the file then the extension freezes VS for a minute and then crashes it. When restarting VS it says this extension is causing problems and should be disabled.

```
// Configure bundling and minification for the project.
// More info at https://go.microsoft.com/fwlink/?LinkId=808241
[
   {
      "outputFileName": "wwwroot/css/site.min.css",

      //Minify all CSS files in the css folder except those already minified
      "inputFiles": [ "wwwroot/css/*(*.css|!(*.min.css)" ]
   },
   {
      "outputFileName": "wwwroot/js/site.min.js",

      //Minify all JS files in the core js folder except those already minified
      "inputFiles": [ "wwwroot/js/*(*.js|!(*.min.js)" ],

      "Optionally" "specify",
      "minification" "options",
      "minify": {
         "enabled": true,
         "renameLocals": true
      },
      // Optionally generate .map file
      "sourceMap": false
   }
]
```

### Steps to recreate
1. Put this file into an aspnet core project.
2. Open the file and interact with it. I tried to delete the spacing above the outputfilename. I made a few other changes prior and saved the file.

### Current behavior
VS hangs for a minute and then crashes. On restart it says the extension should be disabled.

### Expected behavior
Shouldn't hang or crash VS.

**Comment by CoolDadTx** [comment]
I thought it might be related to the missing ) on each of the inputFiles but it crashes even with them there.

**Comment by CoolDadTx** [comment]
Issues seems to be related to the bad JSON that was in the file. The Optionally and minification values were missing colons.

**Comment by jncodex** [comment]
Same problem here with VS Community edition since last year, long time.

**Comment by rockstardev** [comment]
@fordagame thanks for detailed explanation - that was awesome catch!

I too am against message box popping up. So as you say we need just to figure out leaving some kind of info for user. I've just tried running BundlerMinifier with your changes and funny enough, `Logger.Log` actually suffers from same exception (`Due to high risk of deadlock...`) because it tries to access Output Pane.

Ideally we can do something like:

    private static void HandleProcessingException(string configFile, Exception ex)
    {
        BundlerMinifierPackage._dispatcher.BeginInvoke(new Action(() =>
        {
            Logger.Log(ex);

            // some call that will display to user either in status bar or in error list
            // that bundling failed and that user should look at Output Pane for more info
            // look at ErrorMinifyingFile(sender, e) method at the end of this class for guidance
        }));
    }

**Comment by fordagame** [comment]
@rockstardev Good that you note that logger crash as well. I also saw that it crashes, but I didn't check the exception and I just though that it is because it is debugining.

I am not really familiar with this code. Can you explain what is idea BundlerMinifierPackage._dispatcher.BeginInvoke doing? How does this helping?  
If you ask me, I would just remove whole method HandleProcessingException. It initial idea is to include some code, which will show to user that something is wrong with configuration. This is good... if you have any clue what you are showing to user. Thats why actually it opens automatically configuration file. But what if the exception has nothing in common with configuration file? Why we always focus it. Current problem is actually not a problem of configuraion file. So if we want to do this, we need to find better aproach which will handle specific CASE, NOT AN EXCEPTION. And of course to found some way to not cause crash. I guess that this is mean of BundlerMinifierPackage._dispatcher.BeginInvoke(even if I don't understand what is the flow of the code).
The fact that Logger is crashing too, means that actually we should fix Logger, not HandleProcessingException. It is responsibility of logger to not crash in this case.
TL;DR
What I propose is to leave only logger and to find some fix for logger too. But the fix should be applied inside logger. Because on all other places logger will crash as well right?

I actually for myself would even remove this logger, because I have never wanted to see it. I even didn't know until now that it has log. And the most important question is does we really want logger and what kind of logger we want? Do you really check your output for problems with BundlesMinifiers extension?

**Comment by fordagame** [comment]
@rockstardev I have made one more commit: 
https://github.com/fordagame/BundlerMinifier/commit/04a1ffa86586105208f46f6f6dad8fc620fe97c5
Here I remove HandleProcessingException method at all. As well I create Pane in Logger in constructor. Anyway I don't see any point to create this pane first time when someone call logger. Logger may be called in such situation where it can't create Pane (as it happen right now). Everything their is static anyway, so there aren't any point to do it on first call at all. Of course if it is never called, we will waste a bit resources, but not more than every time when it is called to throw exception.

And again. If we need to show configuration file in some situation PLEASE do it with apropriate code, which handle the case, not when exception is thrown!

**Comment by rockstardev** [comment]
@fordagame Dispatcher is what you use to schedule work for UI thread from background thread. Lots of problems we are seeing here in issues are now caused because with switch to `AsyncPackage`. Because of that, we now have background threads triggering work that was previously done on UI threads.

As for silently swallowing exceptions... I disagree ;(. You definitely will survive one way or the other... but for average user we need to display at least some info (minimum is showing that exceptions are thrown). So in my commit https://github.com/madskristensen/BundlerMinifier/pull/357/commits/740e0e8be2ac79339af09f4e9c532a126c7249f8 (co-authored you)... I'm just triggering focus of Output pane. Shouldn't be too distracting, but should give user at least notice that:

1. Some exception is being throw, Bundling is failing
2. Details of the exception that we get so user has starting point on what to address

Last - thanks for your help... glad we could together solve this.

Since pull request (#357) resolves this, I'll be closing this issue now. If you want to continue conversation, let's switch to #351 to unify talk and not split between multiple issues.

**Comment by fordagame** [comment]
Your PR is fixing this issue. I would agree with that.
